includes/wc-product-functions.php

```diff
@@ -200,6 +200,40 @@ function wc_get_featured_product_ids() {
 	return $featured_product_ids;
 }
 
+/**
+ * Populate a batch of rating count lookup table data for products.
+ *
+ * @since 4.0.1
+ * @param int $product Product Object.
+ */
+function wc_get_parent_grouped_id( $product ) {
+
+	global $wpdb;
+
+	$group_ids = array();
+
+	if ( $product->is_type( array( 'simple' ) ) ) {
+
+		$product_id = $product->get_id();
+		$group_data = $wpdb->get_col(
+			$wpdb->prepare( // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.PreparedSQLPlaceholders.QuotedDynamicPlaceholderGeneration
+				"SELECT post_id
+				FROM $wpdb->postmeta
+				WHERE meta_key = '_children' 
+				AND meta_value LIKE %s",
+				'%' . $wpdb->esc_like( $product_id ) . '%'
```

Comment:
This query will not generate the expected result because you are search with "LIKE" without putting any restriction, see:

```
10 is like 10 = true
10 is like 100 = true
10 is like 110 = true
10 is like 101 = true
```

And note that you are searching serialized content, see an example:

```
a:4:{i:0;i:13;i:1;i:14;i:2;i:15;i:3;i:34;}
```

So if you are searching for values like `4` will fetch all grouped products that have 4 children products.
Also you'll fetch all grouped products that have more than 4 children products because all instances of `_children` of those products will have some `i:4;i:XX;` note that `i:4` is the array key.